build_daemon/lib/src/managers/build_target_manager.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:watcher/watcher.dart';
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+import '../../data/build_target_request.dart';
+
+bool _isBlacklistedPath(String filePath, Set<RegExp> blackListedPatterns) {
+  // Ignore the first three directories when filtering as these often
+  // relate to team directories which can contain black listed patterns.
+  var splitPath = filePath.split('/');
+  // TODO(grouma) - We'll want to refactor this so that the WatchEvent stream
+  // provided by the google3 build daemon does this automatically.
+  var path = splitPath.length > 3
+      ? '/${splitPath.sublist(3).join('/')}'
+      : splitPath.join('/');
+  if (blackListedPatterns.any((pattern) => path.contains(pattern))) {
+    return true;
+  }
+  return false;
+}
+
+class BuildTarget {
+  // The target to build.
```

Comment:
Maybe "The name of the target to build" or "The build system identifier for the target"?

Otherwise, how is this different from the object in whole which is also a "BuildTarget"?

Reply:
Updated the comment.